src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+            todoTask.setDescription(strings[2].trim());
+            return todoTask;
+        case "D" :
+            DeadlineTask deadlineTask = new DeadlineTask();
+            if (strings[1].trim().equals("1")) {
+                deadlineTask.setDone(true);
+            } else {
+                deadlineTask.setDone(false);
+            }
+            String[] taskDetails = strings[2].split("\\|");
+            deadlineTask.setDescription(taskDetails[0].trim());
+            deadlineTask.setDate(LocalDateTime.parse(taskDetails[1].trim(), formatter));
+            return deadlineTask;
+        case "E" :
+            EventTask eventTask = new EventTask();
+
+            if (strings[1].trim().equals("1")) {
+                eventTask.setDone(true);
+
+            } else {
+                eventTask.setDone(false);
+            }
+            String[] taskDetails2 = strings[2].split("\\|");
+            eventTask.setDescription(taskDetails2[0].trim());
+            eventTask.setDateTime(LocalDateTime.parse(taskDetails2[1].trim(), formatter));
+            return eventTask;
+        default :
+            return new Task();
+        }
+    }
+
+    /**
+     * This method parse given user command and turns it into and
+     * Command object.
+     *
+     * @param command String User command.
+     * @return Command Return various Command type based on user input.
+     */
+    public static Command parseCommand(String command) {
+        if (command.toLowerCase().equals(Command.EXIT_COMMAND)) {
+            return new ExitCommand();
+        } else if (command.toLowerCase().equals(Command.LIST_COMMAND)) {
+            return new ListCommand();
+        } else if (command.toLowerCase().equals(Command.DONE_ALL_COMMAND)) {
+            return new DoneAllCommand();
+        } else if (command.toLowerCase().contains(Command.DONE_COMMAND)) {
+            return new DoneCommand(command);
+        } else if (command.toLowerCase().contains(Command.TODO_COMMAND)) {
+            return new TodoCommand(command);
+        } else if (command.toLowerCase().contains(Command.DEADLINE_COMMAND)) {
+            return new DeadlineCommand(command);
+        } else if (command.toLowerCase().contains(Command.EVENT_COMMAND)) {
+            return new EventCommand(command);
+        } else if (command.toLowerCase().equals(Command.HELP_COMMAND)) {
+            return new HelpCommand();
+        } else if (command.toLowerCase().contains(Command.DELETE_ALL_COMMAND)) {
+            return new DeleteAllCommand();
+        } else if (command.toLowerCase().contains(Command.DELETE_COMMAND)) {
+            return new DeleteCommand(command);
+        } else if (command.toLowerCase().contains(Command.SHOW_AFTER_COMMAND)) {
+            return new ShowAfterCommand(command);
+        } else if (command.toLowerCase().contains(Command.SHOW_BEFORE_COMMAND)) {
+            return new ShowBeforeCommand(command);
+        } else if (command.toLowerCase().contains(Command.FIND_COMMAND)) {
```

Comment:
In these checks, consider using `startswith` or a more appropriate check as it might find the keyword anywhere within the input.